src/components/ConfirmDialog.js

```diff
@@ -10,7 +10,7 @@ export const ConfirmDialog = ({
 
   const onDiscardFn = () => {
     onDiscard()
-    close()
```

Comment:
this has side-effect, it should only close one level from the dialog! will look close into it!

Reply:
so in general the bug comes from the popup `close` and `show` function, the state doesn't fully update yet but the show/close event is called, so the softkey popup state is mess up with it. the fixed should be considered in https://phabricator.wikimedia.org/T269158